lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -19,6 +19,9 @@ def initialize(queue, input)
       @queue = queue
       @logger = input.logger
       @input = input
+      @metric = @input.metric
+      @peak_connection_count = Concurrent::AtomicFixnum.new(0)
+      @current_connection_count = Concurrent::AtomicFixnum.new(0)
```

Comment:
I think we can derive the value of this by doing `@connections_list.size`, since we are using a *ThreadSafe::Hash*, the `#size` should also be synchronized.